src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas3/DomainEventBuilder.kt

```diff
@@ -98,9 +106,16 @@ class DomainEventBuilder {
             label = departure.moveOnCategory.legacyDeliusCategoryCode ?: "",
           ),
           applicationId = application?.id,
+          applicationUrl = application.toUrl(),
           reasonDetail = null,
         ),
       ),
     )
   }
+
+  private fun TemporaryAccommodationApplicationEntity?.toUrl(): URI? =
+    this?.let { URI(applicationUrlTemplate.replace("#applicationId", it.id.toString())) }
+
+  private fun BookingEntity.toUrl(): URI =
```

Comment:
When we do the referral.submitted event we won't have a booking but will have a referral so this may also need to be optional. Happy to leave that change until later if you'd prefer.

Reply:
I think in this situation we could just not have a `bookingId` field on the event and not call the method.